File: benchmarks/ackermann.java
public class ackermann {
    private static long ackermann(long m, long n) {
        if (m == 0) {
            return n + 1;
        }
        if (n == 0) {
            return ackermann(m - 1, 1);
        }
        return ackermann(m - 1, ackermann(m, n - 1));
    }

    public static void main(String[] args) {
        System.out.println(ackermann(3, 11));
    }
}

File: benchmarks/ackermann.lua
-- FIXME: Will overflow the stack.

function ackermann (m, n)
    if m == 0 then
        return n + 1
    end
    if n == 0 then
        return ackermann(m - 1, 1)
    end
    return ackermann(m - 1, ackermann(m, n - 1))
end

print(ackermann(3, 11))
